src/modules/auth/auth.ts

```diff
@@ -177,6 +214,11 @@ export class Auth {
     };
     const result = await this.#post("/token", payload);
     const { access_token: accessToken, refresh_token: refreshToken } = result;
+    if (!refreshToken) {
```

Comment:
I can think of two cases when this scenario will occur. One, when we are trying to rotate the tokens but the current access token was generated by client credentials flow, so token rotation is not supported. Second, when our SSO server hasn't sent the expected response. For the second case, `this.#post` should throw the error (401, 404 etc. Looks like I missed adding the logic for that. I will create a PR with the proper handling from the `#post` method) so we don't need to do anything. But we should prevent the first case from happening. The `refresh_token` returned by the `await this.getRefreshToken(tokenGetParam)` call at line 212 will be undefined because we don't have a refresh token. So I think we should check first if `await this.getRefreshToken(tokenGetParam)` is giving us a refresh token, and if not, we can throw the error from there (before line 209). The error message in that case can say something like `Cannot rotate the token because of missing refresh token`

Let me know what you think.

Reply:
Agree with throwing an error if refresh token is not returned from `this.getRefreshToken(tokenGetParam)`

I am not sure if I really like the way I implemented it, let me know if this works or if you think there is a better way:

```javascript
    const oldRefreshToken = await this.getRefreshToken(tokenGetParam);
    if (!oldRefreshToken) {
      throw new Error(
        "Cannot refresh access token because the refresh token is missing"
      );
    }
    const payload: TokenRotatePayload = {
      client_id: this.#clientId,
      client_secret: this.#clientSecret,
      refresh_token: oldRefreshToken,
      grant_type: GrantType.REFRESH_TOKEN,
    };
```

Reply:
This looks good.